library/Mockery/Expectation.php

```diff
@@ -216,6 +216,32 @@ protected function _getReturnValue(array $args)
         } elseif (count($this->_returnQueue) > 0) {
             return current($this->_returnQueue);
         }
+
+        $rm = $this->_mock->mockery_getMethod($this->_name);
+        if ($rm && version_compare(PHP_VERSION, '7.0.0-dev') >= 0 && $rm->hasReturnType()) {
+            $type = (string) $rm->getReturnType();
+            switch ($type) {
+                case '':       return;
+                case 'string': return '';
+                case 'int':    return 0;
+                case 'float':  return 0.0;
+                case 'bool':   return false;
+                case 'array':  return [];
+
+                case 'callable':
+                case 'Closure':
+                    return function () {};
+
+                case 'Traversable':
```

Comment:
Iterators are also traversable and we can support them separately when PHP 5.4 is used.